llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp

```diff
@@ -3616,7 +3616,15 @@ class InstAnalyzer {
       hasChain = true;
 
     if (const CodeGenIntrinsic *IntInfo = N.getIntrinsicInfo(CDP)) {
-      ModRefInfo MR = IntInfo->ME.getModRef();
+      // Ignore reads/writes to inaccessible memory. These should not imply
+      // mayLoad/mayStore on the instruction because they are often used to
+      // model dependencies that Machine IR expresses as uses/defs of a
+      // special physical register.
+      ModRefInfo MR = ModRefInfo::NoModRef;
+      for (MemoryEffects::Location Loc : MemoryEffects::locations()) {
+        if (Loc != MemoryEffects::Location::InaccessibleMem)
+          MR |= IntInfo->ME.getModRef();
```

Comment:
Silly typo here. It should have been:
```suggestion
          MR |= IntInfo->ME.getModRef(Loc);
```
Because of this the patch had no effect, and if I fix the typo then it causes some build failures.

I will revert and rethink.